main.go

```diff
@@ -137,15 +139,17 @@ func main() {
 	go func() {
 		fmt.Println("started")
 		for {
+			m.Lock()
```

Comment:
Shouldn't this be inside the `case` definitions?

Reply:
Yes, of cause.